test/Dialect/FIRRTL/parse-basic.fir

```diff
@@ -1113,3 +1113,17 @@ circuit MyModule :     ; CHECK: firrtl.circuit "MyModule" {
     smem testharness : UInt<8>[16] [34359738368]
     node w_addr = UInt<36>(42) @[Cat.scala 31:58]
     write mport MPORT = testharness[w_addr], clock
+
+  module SomeModule:
+    input in: UInt
+    output out: UInt
+    out <= in
+
+    ; Module as identifier
+    ; Test parsing of "module" as an identifier for instance names
+
+  module ModuleAsIdentifier:
+    inst module of SomeModule
+    ; CHECK: %module_in, %module_out = firrtl.instance module
+    module.in <= UInt(1)
+    ; CHECK: firrtl.connect %module_in, %c1_ui : !firrtl.uint, !firrtl.uint
```

Comment:
Mega-nit: This test can be slimmed down to just:

```
module SomeModule:

module ModuleAsIdentifier:
  inst module of SomeModule
```

Reply:
I think I added the rest to check for accessing the instance as Daniel suggested here https://github.com/llvm/circt/pull/4589#issuecomment-1404155876

Reply:
Not sure if that's actually necessary..?

Reply:
Knocking test down a little and using `module` as name of a module too while at it:
```mlir
  ; Module as identifier
  ; Test parsing of "module" as an identifier for instance and module names
  extmodule module:
    input in: UInt

  module ModuleAsIdentifier:
    inst module of module
    ; CHECK: firrtl.instance module interesting_name @module
    module.in <= UInt(1)
```

Leaving the instance test for now, as looks like there's work to be done there (see other comment).